### PR TITLE
setup openreader schema for new archives

### PR DESCRIPTION
**Pull request by tmcgroul**

### CODE REVIEW

File: openreader/src/queryBuilder.ts [hunk 1]
```diff
@@ -465,9 +485,15 @@ export class QueryBuilder {
...
                         break
                     }
-                    case 'list-lookup':
-                        rec[req.alias] = this.toResult(row[req.index], req.children)
+                    case 'list-lookup': {
+                        let rows = row[req.index]
+                        if (rows != null) {
+                            rec[req.alias] = this.toResult(row[req.index], req.children)
+                        } else {
+                            rec[req.alias] = []
```
Comment (tmcgroul) [inline openreader/src/queryBuilder.ts:493]: I think that query builder shouldn't specify default values but i haven't found a better place

Another way is to make default values explicit and extend FieldRequest interface with an optional field for default value
Or just iterate over data returned from querybuilder and search for null values which should be replaced by default value

But both variants are quite ugly

Comment (tmcgroul) [reply openreader/src/queryBuilder.ts:493]: @eldargab

Comment (eldargab) [reply openreader/src/queryBuilder.ts:493]: It's perfectly fine.

Please, reverse `rows != null` to `rows == null`

**Review by eldargab — CHANGES_REQUESTED** [review]
Let's split this PR in two.

First to master branch with a new JSON scalar, as it is useful regardless of new release. Second to this branch with graphql schema for archive.

Apart from that, please provide all necessary tests [here](https://github.com/subsquid/squid/blob/2bdeb1b0c4c2674e4cb8567d48052b42dc251d88/openreader/src/test/scalars.test.ts#L1) demonstrating that new scalar works as intended.

File: openreader/src/gql/scalars/JSON.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import {GraphQLScalarType} from "graphql"
+
+
+export const JSONScalar = new GraphQLScalarType({
+    name: 'JSON',
+    description: 'A scalar that can represent any JSON value',
```
[the remark this answers is not in the document]
Comment (tmcgroul) [reply openreader/src/gql/scalars/JSON.ts:6]: Yes, graphql lib does it fine by itself

**Pull request closed without merging**